extensions/amp-story-panning-media/0.1/amp-story-panning-media.js

```diff
@@ -156,17 +183,74 @@ export class AmpStoryPanningMedia extends AMP.BaseElement {
   }
 
   /** @private */
-  onPageNavigation_() {
-    if (this.isOnActivePage_) {
-      this.animate_();
+  setAnimateTo_() {
+    const x = parseFloat(this.element_.getAttribute('x') || 0);
+    const y = parseFloat(this.element_.getAttribute('y') || 0);
+    const zoom = parseFloat(this.element_.getAttribute('zoom') || 1);
+    const lockBounds = this.element_.hasAttribute('lock-bounds');
+
+    if (!lockBounds) {
+      this.animateTo_ = {x, y, zoom};
+    } else {
+      // Zoom must be set to calculate maxBounds.
+      this.animateTo_.zoom = zoom < 1 ? 1 : zoom;
+
+      this.setMaxBounds_();
+
+      this.animateTo_.x =
+        x > 0
+          ? Math.min(this.maxBounds_.left, x)
+          : Math.max(this.maxBounds_.right, x);
+
+      this.animateTo_.y =
+        y > 0
+          ? Math.min(this.maxBounds_.top, y)
+          : Math.max(-this.maxBounds_.bottom, y);
```

Comment:
```suggestion
    if (!lockBounds) {
      this.animateTo_ = {x, y, zoom};
      return
    }
      // Zoom must be set to calculate maxBounds.
      this.animateTo_.zoom = zoom < 1 ? 1 : zoom;

      this.setMaxBounds_();

      this.animateTo_.x =
        x > 0
          ? Math.min(this.maxBounds_.left, x)
          : Math.max(this.maxBounds_.right, x);

      this.animateTo_.y =
        y > 0
          ? Math.min(this.maxBounds_.top, y)
          : Math.max(-this.maxBounds_.bottom, y);
```

(early return and less indentation, just need to fix the indents of the suggestion)

Reply:
I worry that this makes the codepath harder to read since it's not clearly a branch.
I revised the branch, let me know what you think.